public/app/features/dashboard-scene/scene/setDashboardPanelContext.ts

```diff
@@ -175,23 +175,23 @@ export function getAdHocFilterVariableFor(scene: DashboardScene, ds: DataSourceR
 }
 
 function updateAdHocFilterVariable(filterVar: AdHocFiltersVariable, newFilter: AdHocFilterItem) {
-  // Check if we need to update an existing filter
-  for (const filter of filterVar.state.filters) {
-    if (filter.key === newFilter.key) {
-      filterVar.setState({
-        filters: filterVar.state.filters.map((f) => {
-          if (f.key === newFilter.key) {
-            return newFilter;
-          }
-          return f;
-        }),
-      });
-      return;
-    }
+  // This function handles 'Filter for value' and 'Filter out value' from table cell
+  // We are allowing to add filters with the same key because elastic search ds supports that
+
+  // Update is only required when we change operator and keep key and value the same
+  //   key1 = value1 -> key1 != value1
+  const updateIndex = filterVar.state.filters.findIndex(
```

Comment:
Can we improve the name of this variable? It took me a while to understand what was doing. 

Maybe something like  `filterToReplaceIndex`, `replaceableFilterIndex `, or `operatorUpdateIndex`